### PR TITLE
OpenID support

### LINKED ISSUE #26: OpenID Connect
Hello people, I like the project, but I was wondering if you have any plans to add OpenID Connect to it. 
AFAIK OpenID Connect is built on top of Oauth2.0 to provide authentication, which Oauth2 doesn't provide (only authorization).

According to [this talk](https://youtu.be/996OiexHze0?t=3001) is just a set of 4 extra steps. But I couldn't find any documentation.

Thanks!

### PR DESCRIPTION
**Pull request by aliev**
- [x] Implement `ResponseTypeIdToken`
- [x] Create `get_id_token` which returns (generates) id_token
- [x] nonce query param is required for `id_token` response type
- [x] tests

Closes #26

### CODE REVIEW

**Comment by aliev** [comment]
@Woile @synchronizing I added OpenID support. As an example I used this project: https://oidcdebugger.com/

not sure what else should be added. the id_token must be generated in get_id_token method of BaseDB class

**Comment by Woile** [comment]
Amazing work Ali!

File: src/aioauth/base/database.py [hunk 1]
```diff
@@ -81,6 +81,18 @@ async def create_authorization_code(
...
+    async def get_id_token(
+        self,
+        request: Request,
+        client_id: str,
+        scope: str,
+        response_type: str,
+        redirect_uri: str,
+        nonce: str,
+    ) -> str:
+        """Returns token_id"""
```
Comment (Woile) [inline src/aioauth/base/database.py:93]: for the `database.py` class, I would add comments like:
`"implement for oauth2"` or `"required for openidconnect"`

Comment (aliev) [reply src/aioauth/base/database.py]: @Woile yes make sense. updated :)

**Pull request merged**